### PR TITLE
Add missing default System.Private.Xml test runs

### PR DESCRIPTION
**Pull request by sepidehMS**
Some of the Xml tests did not run in the default CI builds as they passed in non-`netcoreapp1.1` TestTFMs, so I added new runs in System.Private.Xml.Tests.builds for them.

Addresses: https://github.com/dotnet/corefx/issues/12483

cc: @stephentoub @danmosemsft @weshaggard @joperezr

### CODE REVIEW

File: src/System.Private.Xml/tests/System.Private.Xml.Tests.builds [hunk 1]
```diff
@@ -3,65 +3,125 @@
   <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), dir.props))\dir.props" />
   <ItemGroup>
     <Project Include="Readers\CharCheckingReader\System.Xml.RW.CharCheckingReader.Tests.csproj">
+      <TargetGroup>netstandard1.3</TargetGroup>
```
Comment (sepidehMS) [inline src/System.Private.Xml/tests/System.Private.Xml.Tests.builds]: When I try running these tests with `/p:FilterToTestTFM=netcoreapp1.0`, I hit https://github.com/dotnet/coreclr/issues/7607. Is there a way to work around that?

**Comment by stephentoub** [comment]
@dotnet-bot test Outerloop Windows_NT Debug please

**Comment by stephentoub** [comment]
@dotnet-bot test Outerloop Ubuntu14.04 Debug please
@dotnet-bot test Outerloop OSX Debug please

**Comment by stephentoub** [comment]
I've kicked off outer loop runs on several additional OSes... there were failures in the Windows outer loop run:
```
System.Xml.Tests.XmlWriterTestModule.TCDocument
System.Xml.Tests.XmlWriterTestModule.TCFragmentCL
System.Xml.Tests.XmlWriterTestModule.TCNewLineOnAttributes
System.Xml.Tests.XmlWriterTestModule.TCElement
System.Xml.Tests.XmlWriterTestModule.TCWriteAttributes
System.Xml.Tests.XmlWriterTestModule.TCOmitXmlDecl
System.Xml.Tests.XmlWriterTestModule.TCAutoComplete
System.Xml.Tests.XmlWriterTestModule.TCStandAlone
System.Xml.Tests.XmlWriterTestModule.TCWriteNode_XmlReader
System.Xml.Tests.XmlWriterTestModule.TCAttribute
System.Xml.Tests.XmlWriterTestModule.TCCheckChars
```

File: src/System.Private.Xml/tests/System.Private.Xml.Tests.builds (hunk 1 above)
Comment (joperezr) [inline src/System.Private.Xml/tests/System.Private.Xml.Tests.builds]: No, or at least not now. We need to version the mscorlib facade on the CoreCLR package in order to fix this. For now, don't worry about netcoreapp1.0 runs as they are pretty much failing for all projects for the same reason.

File: src/System.Private.Xml/tests/System.Private.Xml.Tests.builds [hunk 2]
```diff
@@ -3,65 +3,125 @@
...
       <OSGroup>Windows_NT</OSGroup>
       <TestTFMs>netcoreapp1.0;netcore50;net46</TestTFMs>
     </Project>
+    <Project Include="Readers\CharCheckingReader\System.Xml.RW.CharCheckingReader.Tests.csproj">
+      <OSGroup>Windows_NT</OSGroup>
+    </Project>
     <Project Include="Readers\CustomReader\System.Xml.RW.CustomReader.Tests.csproj">
+      <TargetGroup>netstandard1.3</TargetGroup>
```
Comment (joperezr) [inline src/System.Private.Xml/tests/System.Private.Xml.Tests.builds]: I would have expected changes on the  actual test .csprojs that set `NugetTargetMoniker` property to `netstandard1.7` now by default, plus changes in the project.jsons to add netstandard1.7 to the frameworks section. Was that not required by your change?

File: src/System.Private.Xml/tests/System.Private.Xml.Tests.builds [hunk 3]
```diff
@@ -3,65 +3,125 @@
   <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), dir.props))\dir.props" />
   <ItemGroup>
     <Project Include="Readers\CharCheckingReader\System.Xml.RW.CharCheckingReader.Tests.csproj">
+      <TargetGroup>netstandard1.3</TargetGroup>
       <OSGroup>Windows_NT</OSGroup>
       <TestTFMs>netcoreapp1.0;netcore50;net46</TestTFMs>
     </Project>
+    <Project Include="Readers\CharCheckingReader\System.Xml.RW.CharCheckingReader.Tests.csproj">
+      <OSGroup>Windows_NT</OSGroup>
```
Comment (joperezr) [inline src/System.Private.Xml/tests/System.Private.Xml.Tests.builds]: I see that in all of the configurations in this .builds file we are passing `OSGroup` to be `Windows_NT`. Is that actually required? E.g. Do we actually have windows-specific tests and we only want to run those?

File: src/System.Private.Xml/tests/System.Private.Xml.Tests.builds (hunk 2 above)
Comment (sepidehMS) [inline src/System.Private.Xml/tests/System.Private.Xml.Tests.builds]: That's correct, as discussed in https://github.com/dotnet/corefx/issues/12483 as well. I'm going to add that to this PR soon.

**Comment by sepidehMS** [comment]
@dotnet-bot test Outerloop Windows_NT Debug please

File: src/System.Private.Xml/tests/System.Private.Xml.Tests.builds (hunk 3 above)
Comment (sepidehMS) [inline src/System.Private.Xml/tests/System.Private.Xml.Tests.builds]: I didn't find any platform-specific tests among these tests. Removed the `OSGroup` property.

File: src/System.Private.Xml/tests/System.Private.Xml.Tests.builds [hunk 4]
```diff
@@ -3,94 +3,104 @@
   <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), dir.props))\dir.props" />
   <ItemGroup>
     <Project Include="Readers\CharCheckingReader\System.Xml.RW.CharCheckingReader.Tests.csproj">
-      <OSGroup>Windows_NT</OSGroup>
+      <TargetGroup>netstandard1.3</TargetGroup>
       <TestTFMs>netcoreapp1.0;netcore50;net46</TestTFMs>
     </Project>
+    <Project Include="Readers\CharCheckingReader\System.Xml.RW.CharCheckingReader.Tests.csproj" />
```
Comment (weshaggard) [inline src/System.Private.Xml/tests/System.Private.Xml.Tests.builds]: For consistency can you please include the default (i.e. the one without any metadata) configurations first?

File: src/System.Private.Xml/tests/Writers/XmlWriterApi/CommonTests.cs [hunk 5]
```diff
@@ -9,8 +9,6 @@
 using System.Text;
 using XmlCoreTest.Common;
 
-[assembly: System.Diagnostics.CodeAnalysis.ExcludeFromCodeCoverage]
```
Comment (weshaggard) [inline src/System.Private.Xml/tests/Writers/XmlWriterApi/CommonTests.cs]: Is this intentional? Does code coverage work for this project now?

File: src/System.Private.Xml/tests/XmlSerializer/Performance/System.Xml.XmlSerializer.Performance.Tests.csproj [hunk 6]
```diff
@@ -6,7 +6,7 @@
     <AppDesignerFolder>Properties</AppDesignerFolder>
     <RootNamespace>System.Xml.XmlSerializer.Performance.Tests</RootNamespace>
     <AssemblyName>System.Xml.XmlSerializer.Performance.Tests</AssemblyName>
-    <NugetTargetMoniker>.NETStandard,Version=v1.3</NugetTargetMoniker>
+    <NugetTargetMoniker Condition="'$(TargetGroup)' == ''">.NETStandard,Version=v1.7</NugetTargetMoniker>
   </PropertyGroup>
   <PropertyGroup Condition="'$(UseContractReferences)' == 'true'" >
     <ProjectJson>ContractReferences/project.json</ProjectJson>
```
Comment (weshaggard) [inline src/System.Private.Xml/tests/XmlSerializer/Performance/System.Xml.XmlSerializer.Performance.Tests.csproj]: Does this project.json file still exist?

**Comment by weshaggard** [comment]
@sepidehMS have you tried building and running these other configurations? Do they currently work?

File: src/System.Private.Xml/tests/System.Private.Xml.Tests.builds (hunk 4 above)
Comment (sepidehMS) [inline src/System.Private.Xml/tests/System.Private.Xml.Tests.builds]: Will do.

File: src/System.Private.Xml/tests/Writers/XmlWriterApi/CommonTests.cs (hunk 5 above)
Comment (sepidehMS) [inline src/System.Private.Xml/tests/Writers/XmlWriterApi/CommonTests.cs]: This `System.Diagnostics.CodeAnalysis.ExcludeFromCodeCoverage` type was previously included from the "Common" directory, but now that I've upgraded this project to `netstandard1.7`, that conflicts with the type `ExcludedFromCodeCoverage` from "System.Diagnostics.CodeAnalysis.dll", which no longer can be applied to assemblies. That's the reason I removed this attribute here.

**Comment by sepidehMS** [comment]
In reply to https://github.com/dotnet/corefx/pull/12742#issuecomment-254687353
@weshaggard I have, but the only configurations in which tests run successfully are `netcoreapp1.1` and `netcore50`. In `netcoreapp1.0` I'm seeing the same problem as all other contracts (can't find some dlls). In `net46` it's partly not finding the assemblies and partly actual test failures. I spent some time yesterday on those and to me it looks like they are happening because the wrong version of the assembly was grabbed.

**Comment by sepidehMS** [comment]
@dotnet-bot test Outerloop Windows_NT Debug please
@dotnet-bot test Outerloop Ubuntu14.04 Debug please
@dotnet-bot test Outerloop OSX Debug please

File: src/System.Private.Xml/tests/System.Private.Xml.Tests.builds [hunk 7]
```diff
@@ -2,102 +2,112 @@
...
+    <Project Include="XmlReader\XmlResolver\System.Xml.RW.XmlSystemPathResolver.Tests.csproj" />
     <Project Include="XmlReader\XmlResolver\System.Xml.RW.XmlSystemPathResolver.Tests.csproj">
       <TargetGroup>netstandard1.3</TargetGroup>
       <TestTFMs>netcoreapp1.0;netcore50;net46</TestTFMs>
     </Project>
-    <Project Include="XmlReader\XmlResolver\System.Xml.RW.XmlSystemPathResolver.Tests.csproj" />
-    <Project Include="XmlSchema\XmlSchemaSet\System.Xml.XmlSchemaSet.Tests.csproj" />
-    <Project Include="XmlSchema\XmlSchemaValidatorApi\System.Xml.XmlSchema.XmlSchemaValidatorApi.Tests.csproj" />
+    <Project Include="XmlSchema\XmlSchemaSet\System.Xml.XmlSchemaSet.Tests.csproj">
+      <OSGroup>Windows_NT</OSGroup>
```
Comment (sepidehMS) [inline src/System.Private.Xml/tests/System.Private.Xml.Tests.builds]: `XmlSchema` and `Xsl` tests need changes before they can run successfully on non-Windows platforms (as they are using file paths and URIs). I created issue https://github.com/dotnet/corefx/issues/12789 for that.

**Comment by sepidehMS** [comment]
PTAL

**Comment by sepidehMS** [comment]
@dotnet-bot test Outerloop Windows_NT Debug please
@dotnet-bot test Outerloop Ubuntu14.04 Debug please
@dotnet-bot test Outerloop OSX Debug please

**Comment by danmosemsft** [comment]
Why the project references to the pkgproj? Otherwise LGTM.

**Comment by sepidehMS** [comment]
In reply to https://github.com/dotnet/corefx/pull/12742#issuecomment-255183239
@danmosemsft  I added those pkgproj references since we want to make sure we are testing the most recent changes in the contract being tested.

**Comment by sepidehMS** [comment]
@dotnet-bot test this please

**Pull request merged**